ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package localproject defines LocalProject interface and implements them on localProject
+
+package localproject
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/converter"
+	ecsclient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/ecs"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/secretsmanager"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/urfave/cli"
+)
+
+const (
+	LocalOutDefaultFileName = "./docker-compose.local.yml"
+	LocalOutFileMode        = os.FileMode(0644) // Owner=read/write, Other=readonly
+	LocalInFileName         = "./task-definition.json"
+)
+
+type LocalProject interface {
+	ReadTaskDefinition() error
+	Convert() error
+	Write() error
+	LocalOutFileName() string
+}
+
+type localProject struct {
+	context          *cli.Context
+	taskDefinition   *ecs.TaskDefinition
+	localBytes       []byte
+	localOutFileName string
+}
+
+func New(context *cli.Context) LocalProject {
+	p := &localProject{context: context}
+	return p
+}
+
+func (p *localProject) TaskDefinition() *ecs.TaskDefinition {
+	return p.taskDefinition
+}
+
+func (p *localProject) LocalOutFileName() string {
+	return p.localOutFileName
+}
+
+func (p *localProject) ReadTaskDefinition() error {
+	arn := p.context.String(flags.TaskDefinitionArnFlag)
+	filename := p.context.String(flags.TaskDefinitionFileFlag)
+
+	if arn != "" && filename != "" {
```

Comment:
FYI: @iamhopaul123 is also making modifications for ECS local :) he needed this same logic in `local ps` and `local down` so it was moved to a function here: https://github.com/aws/amazon-ecs-cli/pull/789/files#diff-d09e4b573b1ca106569f2a8dfcad6cd0R50